hypothesis-python/tests/cover/test_constants_ast.py

```diff
@@ -234,6 +236,38 @@ def test_ignores_ast_parse_error(tmp_path):
 @given(constants_classes)
 def test_constant_visitor_roundtrips_string(constants):
     # our files in storage_directory("constants") rely on this roundtrip
-    visitor = ConstantVisitor()
+    visitor = ConstantVisitor(limit=True)
     visitor.visit(ast.parse(str(set(constants))))
     assert visitor.constants == constants
+
+
+def test_too_many_constants():
+    visitor = ConstantVisitor(limit=True)
+    # start at n=1000 to avoid ConstantVisitor ignoring small integers
+    s = "; ".join(
+        f"n = {i}" for i in range(1000, 1000 + ConstantVisitor.CONSTANTS_LIMIT + 1)
+    )
+    # visitor should raise on too many constants
+    with pytest.raises(TooManyConstants):
+        visitor.visit(ast.parse(s))
+
+    # and also _constants_from_source should return empty on too many constants
+    assert _constants_from_source(s, limit=True) == Constants()
+
+    # but it parses fine with limit=False
+    visitor = ConstantVisitor(limit=False)
+    visitor.visit(ast.parse(s))
+    assert _constants_from_source(s, limit=False) == Constants(
+        integers=set(range(1000, 1000 + ConstantVisitor.CONSTANTS_LIMIT + 1))
+    )
+
+
+def test_module_too_large(tmp_path):
+    module = ModuleType("module_too_large")
+
+    large_file = tmp_path / "large_file.py"
+    large_file.write_text("a = 11231783\n\n", encoding="utf-8")
+    large_file.write_text("#" * (512 * 1024 + 1), encoding="utf-8")
```

Comment:
I believe the second `write_text()` call overwrites the first; we should instead concat the strings before writing.

Also, let's assert that constants_from_module finds the constant if limit=False.